interfaces/portal/src/app/utils/locale.ts

```diff
@@ -1,36 +1,52 @@
 import { isDevMode } from '@angular/core';
 
-import { LanguageEnum } from '@121-service/src/shared/enum/language.enums';
+import { UILanguage } from '@121-service/src/shared/enum/ui-language.enum';
 
+import { uiLanguageToLanguageNames } from '~/services/get-registration-preferrred-language-name.service';
 import { environment } from '~environment';
 
+/**
+ * "locale" in this file always refers to Angular locale IDs, e.g. "en-GB",
+ * "fr", "nl", etc.
+ *
+ * We convert these into UILanguage values where we communicate through the
+ * browser or emails we send to users of the portal.
+ *
+ * Registrations have a preferredLanguage field which we do **not** convert to
+ * UILanguage values.
+ */
+
+// TODO: rename this to selectedLocale. Will require a migration.
 const LOCAL_STORAGE_LOCALE_KEY = 'preferredLanguage';
 
 // NOTE: Make sure to align these languages with ALL_AVAILABLE_LOCALES in '_all_available-locales.mjs'
 export enum Locale {
-  ar = 'ar',
+  /* eslint-disable @typescript-eslint/prefer-literal-enum-member -- emphasize the relationship between this enum and UILanguage */
+  ar = UILanguage.ar,
   en = 'en-GB', // this has to be en-GB otherwise angular locale stuff doesn't work
-  es = 'es',
-  fr = 'fr',
-  nl = 'nl',
-  sk = 'sk',
+  es = UILanguage.es,
+  fr = UILanguage.fr,
+  nl = UILanguage.nl,
+  sk = UILanguage.sk,
+  /* eslint-enable @typescript-eslint/prefer-literal-enum-member -- emphasize the relationship between this enum and UILanguage */
 }
 
+const localeToUILanguageMap: Record<Locale, UILanguage> = {
+  [Locale.ar]: UILanguage.ar,
+  [Locale.en]: UILanguage.en,
```

Comment:
is it correct that here you do use the enum and not 'en-GB'? Otherwise I'm not following.